lib/src/neu/generators/json-schema/json-schema-type-util.ts

```diff
@@ -0,0 +1,200 @@
+      };
+    case TypeKind.DATE:
+      return { type: "string", format: "date" };
+    case TypeKind.DATE_TIME:
+      return { type: "string", format: "date-time" };
+    case TypeKind.STRING:
+      return {
+        type: "string"
+      };
+    case TypeKind.STRING_LITERAL:
+      return {
+        type: "string",
+        const: type.value
+      };
+    case TypeKind.FLOAT:
+    case TypeKind.DOUBLE:
+      return {
+        type: "number"
+      };
+    case TypeKind.FLOAT_LITERAL:
+      return {
+        type: "number",
+        const: type.value
+      };
+    case TypeKind.INT32:
+    case TypeKind.INT64:
+      return {
+        type: "integer"
+      };
+    case TypeKind.INT_LITERAL:
+      return {
+        type: "integer",
+        const: type.value
+      };
+    case TypeKind.OBJECT:
+      return type.properties.reduce<JsonSchemaObject & { required: string[] }>(
+        (acc, property) => {
+          if (!property.optional) {
+            acc.required.push(property.name);
+          }
+          acc.properties[property.name] = typeToJsonSchemaType(
+            property.type,
+            objectAdditionalProperties
+          );
+          return acc;
+        },
+        {
+          type: "object",
+          properties: {},
+          required: [],
+          additionalProperties: objectAdditionalProperties
+        }
+      );
+    case TypeKind.ARRAY:
+      return {
+        type: "array",
+        items: typeToJsonSchemaType(
+          type.elementType,
+          objectAdditionalProperties
+        )
+      };
+    case TypeKind.UNION:
+      const elements = type.types;
+      if (areBooleanLiteralTypes(elements)) {
+        return {
+          type: "boolean",
+          enum: elements.map(e => e.value)
+        };
+      } else if (areStringLiteralTypes(elements)) {
+        return {
+          type: "string",
+          enum: elements.map(e => e.value)
+        };
+      } else if (areFloatLiteralTypes(elements)) {
+        return {
+          type: "number",
+          enum: elements.map(e => e.value)
+        };
+      } else if (areIntLiteralTypes(elements)) {
+        return {
+          type: "integer",
+          enum: elements.map(e => e.value)
+        };
+      } else {
+        // Guaranteed oneOf
+
+        const oneOfElements = elements
+          .filter(isNotLiteralType)
+          .map(t => typeToJsonSchemaType(t, objectAdditionalProperties));
+
+        const booleanLiterals = elements.filter(isBooleanLiteralType);
+        switch (booleanLiterals.length) {
+          case 0:
+            break;
+          case 1:
+            oneOfElements.push(typeToJsonSchemaType(booleanLiterals[0]));
+            break;
+          default:
+            oneOfElements.push({
+              type: "boolean",
+              enum: booleanLiterals.map(b => b.value)
+            });
+        }
+
+        const stringLiterals = elements.filter(isStringLiteralType);
+        switch (stringLiterals.length) {
```

Comment:
Looks like this logic is repeated across this function, I'd consider refactoring it - e.g. create some function that returns an array of `oneOfElements`.